### DISCUSSION TITLE
How to use workflow env variables in job ifs?

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#27144] [Actions]
<pre><code>env:
  IS_PUSH: github.event_name == 'push'

jobs:
  ci:
    if: ${{ env.IS_PUSH }}
</code></pre>
<p>This returns an error: “Unrecognized named-value: ‘env’”</p>
<pre><code>env:
  IS_PUSH: github.event_name == 'push'

jobs:
  ci:
    steps:
      - if: ${{ env.IS_PUSH }}
        uses: mathieudutour/github-tag-action@v5
</code></pre>
<p>This works.</p>
<p>According to the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#env" rel="noopener nofollow ugc">GitHub Docs</a>:</p>
<blockquote>
<p>A map of environment variables that are available to <strong>all jobs and steps</strong> in the workflow.</p>
</blockquote>
<p>What am I missing ?<br>
Also what is the object name for the workflow context and is it accessible i.e. workflow.env.variable ?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>There’s a conflicting statement about the env context:</p>
<blockquote>
<p>You can only use the env context in the value of the with and name keys, or <ins>in a step’s if conditional</ins>.</p>
</blockquote>
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#env-context">https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#env-context</a></p>
<p>The other section might need to be changed to:</p>
<blockquote>
<p>A map of environment variables that are available to <strong>the steps of all jobs</strong> in the workflow.</p>
</blockquote>

**Comment by aleks-ivanov** [comment]
<p>Thank you.</p>
<p>It definitely works the way it is explained in the link you provided.</p>

**Comment by Simran-B** [comment]
<p>My PR to make clear that the env context is only available in steps just landed: <a href="https://github.com/github/docs/commit/ad5c362c63ae9170b62d403a97fdfa21fd39366b" class="inline-onebox" rel="noopener nofollow ugc">Actions: env context cannot be used in job.&lt;job_id&gt;.if (#1803) · github/docs@ad5c362 · GitHub</a></p>
